### DISCUSSION TITLE
Let's discuss federal website standards

### DISCUSSION THREAD

**Discussion by michelle-rago** [GSA-TTS/federal-website-standards#111] [Announcements]
## Who can contribute

These discussions are public. We welcome participation from federal agency staff, folks who work on federal websites or digital services, and other stakeholders. 

Participating in GitHub discussions requires a GitHub account. If you don’t want to participate on GitHub, email your feedback to website.standards@gsa.gov. 

### Contribute to a discussion about a standard

Review the discussions and give your feedback on a standard. Share expertise or research that will advance the discussion.

### Propose a new standard

First, check our [standards project board](https://github.com/orgs/GSA-TTS/projects/48/views/13) to make sure your idea isn’t there already. If it’s not there, start a new discussion with your idea. Include this information:
- Describe the standard
- Describe how the standard would improve the user experience on federal websites

### Share an idea

Share an idea about the standards website or about the standards themselves.

### Vote for comments or ideas that you support

Use the upvote feature to show your support for comments or ideas.

## Follow our community guidelines

All discussions must follow the [GSA Technology Transformation Services code of conduct](https://handbook.tts.gsa.gov/about-us/code-of-conduct/).

**Comment by lukefretwell** [comment]
Nothing substantive to add at the outset. As a member of the public who's not involved in working on federal websites, **I just want to share my appreciation** that a) this initiative is happening and b) the multiple ways that you're working in the open to facilitate it (repo, project board, discussions). 

Thank you! 🇺🇸

**Reply by michelle-rago** [reply]
Thank you, Luke!

**Comment by thompsondc** [comment]
Recommend any standard be provided in pure HTML/CSS/JS package as standalone for implementation.  It is reasonable to expect any agency to be able to "fit" the package to their environment, however the [Federal Government Banner](https://standards.digital.gov/standards/banner/) (pending standard) only has two methods of compliance:
1. Using the USWDS, which can only be installed as an NPM package (not only relying upon the ability and authority to use NPM, but also installing the entire USWDS); or 
2. Complying with the standard's acceptance criteria, without specific guidance (or code) to achieve compliance.
While adoption of the USWDS is ideal, it shouldn't be an indirect requirement of compliance with a singular federal website standard.

**Reply by rphillippi** [reply]
Hi, We'll look into that as a future update. For now, a couple items of note:
1. USWDS is currently updating the system to move to web components. You can learn more about that on their [November Monthly Call](https://designsystem.digital.gov/about/monthly-calls/#june-2024-a-first-look-at-u-s-web-design-system-web-components).
2. You could build your own as USWDS is not required to be in compliance either by leveraging the CSS/HTML code from USWDS or writing it yourself. It is encouraged to use USWDS but not required.

**Reply by thompsondc** [reply]
Regarding your #2 - that's what I was referring to, in my #2.  As you stated; the standard only encourages using USWDS, however it isn't really possible to "leverage the CSS/HTML code from USWDS", as it is written in SASS and compiled using NPM.  Because the CSS can't be extracted without the entire USWDS package, it needs to be manually written by tracing the dependencies.
My recommendation, is that before a standard is moved to the "Pending" status; if implementation without the entire USWDS package would require significant agency resources to "self-create", that GSA (as the steward of the standards) provide a "clean" method of implementation.
While some standards would be more adaptable to the environment of each website; the purpose of the Federal Government Banner is to convey the trustworthiness of .gov and .mil sites - by showing the specific banner.  Given the intent of the standard, uniformity in implementation is ideal; so, leaving each agency to implement via a self-created method will likely defeat the purpose.

**Reply by rphillippi** [reply]
Thanks for your feedback. We'll consider that as we move forward.
